lib/engine/game/base.rb

```diff
@@ -1261,11 +1265,20 @@ def priority_deal_player
         end
       end
 
-      def reorder_players
-        player = @players.reject(&:bankrupt)[@round.entity_index]
+      def next_sr_position(entity)
+        player_order = @round.current_entity&.player? ? @round.pass_order : @players
+        player_order.reject(&:bankrupt).index(entity)
+      end
 
-        @players.rotate!(@players.index(player))
-        @log << "#{player.name} has priority deal"
+      def reorder_players
+        case self.class::NEXT_SR_PLAYER_ORDER
+        when :after_last_to_act
+          player = @players.reject(&:bankrupt)[@round.entity_index]
+          @players.rotate!(@players.index(player))
+        when :first_to_pass
+          @players = @round.pass_order if @round.pass_order.any?
```

Comment:
what happens if there is no pass order? it's ok not to set players?

Reply:
If there is no pass order, then there wasn't a round for players to pass, so the order would stay the same.